packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,360 @@
+import { TransactionAnalyzer } from "./transactionAnalyzer";
+import { BtcTransactionTemplate } from "./btcTransactionTemplate";
+import { Network } from "@caravan/bitcoin";
+import {
+  BtcTxInputTemplate,
+  BtcTxOutputTemplate,
+} from "./btcTransactionComponents";
+import { Satoshis, TxOutputType, UTXO, FeeBumpStrategy } from "./types";
+
+/**
+ * Creates a cancel Replace-By-Fee (RBF) transaction.
+ *
+ * This function creates a new transaction that double-spends at least one input
+ * from the original transaction, effectively cancelling it by sending all funds
+ * to a new address minus the required fees.
+ *
+ * @param {string} originalTx - The hex-encoded original transaction to be replaced.
+ * @param {UTXO[]} availableInputs - Array of available UTXOs, including the original transaction's inputs.
+ * @param {string} cancelAddress - The address where all funds will be sent in the cancellation transaction.
+ * @param {Network} network - The Bitcoin network being used (e.g., mainnet, testnet).
+ * @param {number} dustThreshold -  The dust threshold in satoshis. Outputs below this value are considered
+ "dust" and may not be economically viable to spend. This is used in Default Bitcoin Core value is 546 satoshis .
+ * @param {string} scriptType - The type of script used for the transaction (e.g., P2PKH, P2SH, P2WSH).
+ * @param {number} requiredSigners - The number of required signers for the multisig setup.
+ * @param {number} totalSigners - The total number of signers in the multisig setup.
+ * @returns {string} The base64-encoded PSBT of the cancel RBF transaction.
+ * @throws {Error} If RBF is not possible or if the transaction creation fails.
+ *
+ * @example
+ * const cancelTx = createCancelRbfTransaction(
+ *   originalTxHex,
+ *   availableUTXOs,
+ *   'bc1q...', // cancel address
+ *   Network.MAINNET,
+ *   'P2WSH',
+ *   2, // required signers
+ *   3  // total signers
+ * );
+ *
+ * @see https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki BIP 125: Opt-in Full Replace-by-Fee Signaling
+ * @see https://developer.bitcoin.org/devguide/transactions.html#locktime-and-sequence-number Bitcoin Core's guide on locktime and sequence numbers
+ */
+export const createCancelRbfTransaction = (
+  originalTx: string,
+  availableInputs: UTXO[],
+  cancelAddress: string,
+  network: Network,
+  dustThreshold: number,
+  scriptType: string,
+  requiredSigners: number,
+  totalSigners: number,
+): string => {
+  // Step 1: Analyze the original transaction
+  const txAnalyzer = new TransactionAnalyzer({
+    txHex: originalTx,
+    network,
+    targetFeeRate: 0, // We'll use the analyzer's recommendation
+    absoluteFee: 0, // This will be calculated by the analyzer
```

Comment:
So it looks like these basically aren't needed as options to instantiate the analyzer right?

Reply:
although fee might be necessary since you can't infer from the txHex. I'm not sure the absoluteFee can be calculated by the analyzer without the original inputs.

Reply:
Yes ... I missed that totally .

> So it looks like these basically aren't needed as options to instantiate the analyzer right?

only for this usecase we have `targetFeeRate : 0` but if you are using it independently then these are important options to play with

Reply:
As I stated it above ... that targetFeeRate  ... is a very important param for our analyzer to give the new estimated rbf/cpfp rates ...